### ISSUE TITLE
Expand API

### ISSUE THREAD

**Issue by LemmoTresto** [LemmoTresto/LemonMobCoins#9]
- Add event bus
- Add more functionality to the current API by allowing more stuff to happen.

**Comment by LemmoTresto** [comment]
As I've been asked what exactly needs to be done:

**We need to add an event bus**
Options for this are:

- Create our own:
- Use google's (https://github.com/google/guava/wiki/EventBusExplained)
- Or use a different library such as (https://github.com/KyoriPowered/event)

We will need to add events for when the GUI is opened, when someone receives coins (different types as the API also states increment, deduct etc) on reload and a lot more. 

**Next to the event bus we also need to expand the methods and what you can do with the API**
With this I mean we need to allow something hooking into LemonMobCoins to be able to control the GUI for example or the price of things etc. I suggest asking if you have an idea of adding something of this because not everything should be able to be accessed using the API.
In other words make it an API that allows you to do many things with the plugin than just getting and setting coins.
